dask/array/utils.py

```diff
@@ -181,7 +179,10 @@ def allclose(a, b, equal_nan=False, **kwargs):
         return np.allclose(a, b, equal_nan=equal_nan, **kwargs)
     if equal_nan:
         return a.shape == b.shape and all(
-            np.isnan(b) if np.isnan(a) else a == b for (a, b) in zip(a.flat, b.flat)
+            # NOTE: use `pd.isna` instead of `np.isnan` to also handle
+            # pandas NA values which could slip in
+            pd.isna([a, b]).all() if pd.isna([a, b]).any() else a == b
+            for (a, b) in zip(a.flat, b.flat)
```

Comment:
@jrbourbeau how do you feel about switching `np.isnan` to `pd.isna` in array's `assert_eq`? It behaves the same on NaNs as NumPy (in fact just uses `np.isnan` on array-likes), just is also aware of pandas's various NA values. Don't think it'll actually affect anything, but maybe it's overstepping the scope of the array module?